onnx/common/ir_pb_converter.cc

```diff
@@ -288,6 +291,17 @@ std::unique_ptr<Graph> graphProtoToGraph(const ONNX_NAMESPACE::GraphProto& gp, b
     }
   }
 
+  for (int i = 0; i < gp.initializer_size(); ++i) {
+    auto init = tensorProtoToTensor(gp.initializer(i));
+    g->addInitializer(init, init.name());
+    // If ir_version >= 4, initializer does not have to be included in input
+    // Create a Value from initializer by addInitializerNode
+    // and save it into value_by_name_of for later use (node input)
+    if (ir_version >= 4) {
+      value_by_name_of[init.name()] = g->addInitializerNode(init, init.name());
```

Comment:
did you test this?
This is what happens in addInitializer: 
```
initializers_.push_back(std::move(initializer));
initializer_names_.push_back(std::move(name));
```
you are using initializer and name after move.

Reply:
I think it should be okay since addInitializer's parameters are passed by-value, so no impact on the values here?

Reply:
@jcwchen : thanks for the explanation (in the other thread). In ir_version >= 4, initializer may or may not be in the input list. I think you need to check here whether initializer X already has a value (because it is in the input list) also. We should addInitializerNode only if X is not an input. (If you want value_by_name_of for all initializers, setting it should probably be moved out of the conditional. But do we need it?

Reply:
Eg., change the condition to
```cpp
   if ( (ir_version >= 4) && (value_by_name_of.count(init.name()) > 0)) {
```